object.go

```diff
@@ -754,19 +793,28 @@ func (obs *obs) UpdateMeta(name string, meta *ObjectMeta) error {
 	if meta == nil {
 		return ErrBadObjectMeta
 	}
-	// Grab meta info.
+
+	// Grab info.
 	info, err := obs.GetInfo(name)
 	if err != nil {
 		return err
 	}
-	// Copy new meta
-	info.ObjectMeta = *meta
-	mm := NewMsg(fmt.Sprintf(objMetaPreTmpl, obs.name, sanitizeName(meta.Name)))
-	mm.Data, err = json.Marshal(info)
-	if err != nil {
-		return err
+
+	// Update Meta prevents update of ObjectMetaOptions (Link, ChunkSize)
+	// These should only be updated internally when appropriate.
+	info.Name = meta.Name
+	info.Description = meta.Description
+	info.Headers = meta.Headers
+
+	if err = publishMeta(obs, meta.Name, info); err != nil { return err }
+
+	// did the name of this object change? We just stored the meta under the new name
+	// so delete the meta from the old name
+	if err == nil && name != meta.Name {
+		metaSubj := fmt.Sprintf(objMetaPreTmpl, obs.name, name)
```

Comment:
I think you have a helper for this no?

Reply:
yeah this should be the toDefaultMetaSubject since the subject will be used as data in a JS.API call versus publishing